hyperspy/learn/mva.py

```diff
@@ -1222,13 +1248,54 @@ def undo_treatments(self):
         self.data[:] = self._data_before_treatments
         del self._data_before_treatments
 
+    def _estimate_elbow_position(self,error_estimate):
+        """
+        Estimate the elbow position of a curve
+        Used to estimate the no of significant components in PCA variance ratio
+        plot or other "elbow" type curves
+
+        Parameters
+        ----------
+        error_estimate : array
+
+        Returns
+        -------
+        elbow position : int
+        
+        Index of the elbow position (+1)
+        
+        """
+        maxpoints = min(20,len(error_estimate)-1)
+        # Find a line between first and last point 
+        # With a classic elbow scree plot the line from first to last
+        # more or less defines a triangle 
+        # The elbow should be the point which is the
+        # furthest distance from this line
+        #
+        y2 = np.log(error_estimate[maxpoints])
+        x2 = maxpoints
+        y1 = np.log(error_estimate[0])
+        x1 = 0
+        # loop 
+        distance = np.zeros((maxpoints))
+        for i in range(maxpoints):
```

Comment:
I'll add here that it would be better to do this with a vector calculation (https://stackoverflow.com/a/46470401/1435788), rather than looping through the array, but since it's such a small one-dimensional array, there'e probably no harm in leaving it as a loop.

Reply:
If it's ok I'll leave it as is for now as I think it's negligible computationally compared to the decomposition process in the majority of cases.

Reply:
I've merged and edited the changes and comments from @jat255  and @ericpre